Core/src/Surfaces/AnnulusBounds.cpp

```diff
@@ -172,19 +174,35 @@ bool Acts::AnnulusBounds::inside(const Vector2& lposition, double tolR,
   return true;
 }
 
-bool Acts::AnnulusBounds::inside(const Vector2& lposition,
-                                 const BoundaryCheck& bcheck) const {
-  // locpo is PC in STRIP SYSTEM
-  if (bcheck.type() == BoundaryCheck::Type::eAbsolute) {
-    return inside(lposition, bcheck.tolerance()[eBoundLoc0],
-                  bcheck.tolerance()[eBoundLoc1]);
+bool Acts::AnnulusBounds::inside(
+    const Vector2& lposition,
+    const BoundaryTolerance& boundaryTolerance) const {
+  if (boundaryTolerance.isInfinite()) {
+    return true;
+  }
+
+  if (boundaryTolerance.isNone()) {
+    return inside(lposition, 0., 0.);
+  }
+
+  if (auto absoluteBound = boundaryTolerance.asAbsoluteBoundOpt();
```

Comment:
So this `asAbsoluteBoundOpt` is sort of the alternative to returning the variant and visiting it?

Reply:
Right this does some logic inside which handles the cases of `None`, `AbsoluteBound`, `AbsoluteCartesian` in case we are cartesian anyways